src/GraphQL/Types/IAbstractGraphType.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GraphQL.Types
+{
+    public interface IAbstractGraphType
+    {
+        Func<object, ObjectGraphType> ResolveType { get; set; }
+
+        List<ObjectGraphType> PossibleTypes { get; }        
```

Comment:
You can keep the interface, though add the `Add` method to the interface.  You can then also have a base implementation of that interface to remove the duplication.  That is a common pattern.